examples/gno.land/r/gnoland/users/v1/render.gno

```diff
@@ -0,0 +1,146 @@
+package v1
+
+import (
+	"std"
+	"strconv"
+
+	"gno.land/p/demo/avl/pager"
+	"gno.land/p/demo/ufmt"
+	"gno.land/p/moul/md"
+	"gno.land/p/moul/realmpath"
+	"gno.land/p/moul/txlink"
+
+	"gno.land/r/demo/profile"
+	"gno.land/r/sys/users"
+)
+
+var (
+	connectUrlBase       = "https://gno.studio/connect/view/gno.land/r/gnoland/users/v1?network=" + std.GetChainID()
+	connectRegisterUrl   = connectUrlBase + "#Register"
+	connectUpdateNameUrl = connectUrlBase + "#UpdateName"
+)
+
+func Render(path string) string {
+	req := realmpath.Parse(path)
+
+	if req.Path == "" {
+		return renderHomePage(path)
+	}
+
+	// Otherwise, render the user page
+	return renderUserPage(req.Path)
+}
+
+func renderHomePage(path string) string {
+	var out string
+
+	out += "# gno.land user registry\n"
+
+	if paused {
+		out += md.HorizontalRule()
+		out += md.H2("This realm is paused.")
+		out += md.Paragraph("Check out [`gno.land/r/gnoland/users`](/r/gnoland/users) for newer versions of the registry.")
+		out += md.HorizontalRule()
+	}
+
+	out += renderIntroParagraph()
+
+	out += md.H2("User list")
+
+	p := pager.NewPager(users.GetReadOnlyNameStore(), 20, false)
+	page := p.MustGetPageByPath(path)
+	for _, item := range page.Items {
+		if item.Value == nil {
+			continue
+		}
+
+		data := item.Value.(*users.UserData)
+
+		// Skip previous names
+		if item.Key != data.Name() {
+			continue
+		}
+
+		out += ufmt.Sprintf("- User [%s](/r/gnoland/users/v1:%s)\n", md.Bold(item.Key), item.Key)
+	}
```

Comment:
Even though I'm glad you're using this pattern that I was proud of (thank you, @leohhhn), I believe this realm will quickly become unusable because it will contain:

-  aaa000
-  aaa001
-  aaa002
-  etc.

It makes more sense to either print statistics or eventually set up an efficient ring buffer for the last N entries (we don't have a good data structure yet, so let's create a p/ for this :p). In the meantime, just store the last created *UserData entry.

Reply:
https://github.com/gnolang/gno/pull/3736